### PR TITLE
Fix missing default query configuration

### PR DESCRIPTION
**Pull request by Arkatufus**
When using HOCON configuration, default query configuration were not injected properly

## Changes

* Inject default query configuration into SqlReadJournalProvider
* Standardize SqlReadJournalProvider
* Add sample project

### CODE REVIEW

File: src/Akka.Persistence.Sql/Query/SqlReadJournal.cs [hunk 1]
```diff
@@ -68,7 +67,7 @@ public class SqlReadJournal :
             _mat = Materializer.CreateSystemMaterializer(
                 context: system,
                 settings: ActorMaterializerSettings.Create(system),
-                namePrefix: $"l2db-query-mat{configPath}");
+                namePrefix: $"l2db-query-mat-{Guid.NewGuid():N}");
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Sql/Query/SqlReadJournal.cs:70]: What was the fix here?

**Review by Aaronontheweb — COMMENTED** [review]
Not sure I understand the source of the fix - could you help me @Arkatufus ?

File: src/Akka.Persistence.Sql/Query/SqlReadJournal.cs (hunk 1 above)
Comment (Arkatufus) [reply src/Akka.Persistence.Sql/Query/SqlReadJournal.cs:70]: That was not the fix, the `SqlReadJournal` is a bit different in that it creates a materializer, which none of our other read journal does.

It tries to create a single materializer by appending the "config path".

The problem is that "config path" were never being passed into the provider, the provider was created through reflection magic and none of the activator code passes in the "config path".

So I just remove that ambiguous code and replace it with a GUID

File: src/Akka.Persistence.Sql/Query/SqlReadJournalProvider.cs [hunk 2]
```diff
@@ -12,29 +12,17 @@ namespace Akka.Persistence.Sql.Query
...
-
-        public SqlReadJournalProvider(
-            ExtendedActorSystem system,
-            Configuration.Config config,
-            string configPath)
-        {
-            _system = system;
-            _config = config;
-            _configPath = configPath;
+            _config = config.WithFallback(SqlPersistence.DefaultQueryConfiguration);
```
Comment (Arkatufus) [inline src/Akka.Persistence.Sql/Query/SqlReadJournalProvider.cs:22]: This is the actual fix

Comment (Aaronontheweb) [reply src/Akka.Persistence.Sql/Query/SqlReadJournalProvider.cs:22]: Got it, ok thank you.

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**